src/applications/disability-benefits/526EZ/actions/index.js

```diff
@@ -1,64 +1,157 @@
+      status = PRESTART_STATUSES.expired;
+      expirationDate = getLatestTimestamp(expiredList.map(itf => itf.expirationDate));
+      dispatch(setPrestartData({ previousExpirationDate: expirationDate }));
+      // If the user has an active ITF, set retrieved status
+    } else {
+      status = PRESTART_STATUSES.retrieved;
+      expirationDate = activeList[0].expirationDate;
+      dispatch(setPrestartData({ currentExpirationDate: expirationDate }));
+    }
+  }
+  dispatch(setPrestartStatus(status));
+  return status;
+};
+
+export const handleCheckFailure = (error, hasSavedForm, dispatch) => {
+  const status = hasSavedForm ? PRESTART_STATUSES.notRetrievedSaved : PRESTART_STATUSES.notRetrievedNew;
+  dispatch(setPrestartStatus(status));
+  return status;
+};
+
+export function checkITFRequest(dispatch, hasSavedForm) {
+  return apiRequest(
+    '/intent_to_file',
+    null,
+    ({ data }) => handleCheckSuccess(data, dispatch),
+    ({ errors }) => handleCheckFailure(errors, hasSavedForm, dispatch)
+  );
+}
+
+const delay = (func, wait, args) => setTimeout(() => func(args), wait);
+const fakeITFRequest = (url, options, success) => delay(success, 1000, existingITFData);
+
+export function mockCheckITFRequest(dispatch, hasSavedForm) {
+  return fakeITFRequest(
+    '/intent_to_file',
+    null,
+    ({ data }) => handleCheckSuccess(data, dispatch),
+    ({ errors }) => handleCheckFailure(errors, hasSavedForm, dispatch)
+  );
+}
+
+export const handleSubmitSuccess = (data, successStatus, dispatch) => {
+  const expirationDate = data.attributes.intentToFile.expirationDate;
+  dispatch(setPrestartData({ currentExpirationDate: expirationDate }));
+  dispatch(setPrestartStatus(successStatus));
+};
+
+export const handleSubmitFailure = (error, errorStatus, dispatch) => {
+  dispatch(setPrestartStatus(errorStatus));
+};
 
+export function submitITFRequest(dispatch, successStatus, errorStatus) {
+
+  return apiRequest(
+    '/intent_to_file/compensation',
+    { method: 'POST' },
+    ({ data }) => handleSubmitSuccess(data, successStatus, dispatch),
+    ({ errors }) => handleSubmitFailure(errors, errorStatus, dispatch)
+  );
+}
+
+export function verifyIntentToFile(hasSavedForm) {
+  return async (dispatch) => {
+    let submitSuccessStatus;
+    let submitErrorStatus;
+    dispatch(setPrestartStatus(PRESTART_STATUSES.pending));
+
+    const existingITFStatus = await checkITFRequest(dispatch, hasSavedForm);
+
+    if (!prestartPendingStatuses.has(existingITFStatus)) {
+      return;
+    }
+
+    if (existingITFStatus === PRESTART_STATUSES.none) {
+      submitSuccessStatus = PRESTART_STATUSES.created;
+      submitErrorStatus = PRESTART_STATUSES.notCreated;
+    } else if (existingITFStatus === PRESTART_STATUSES.expired) {
+      submitSuccessStatus = PRESTART_STATUSES.renewed;
+      submitErrorStatus = PRESTART_STATUSES.notRenewed;
+    }
+    submitITFRequest(dispatch, submitSuccessStatus, submitErrorStatus);
```

Comment:
Do we always want to `POST` an ITF when we `verifyIntentToFile`?

Reply:
We only reach this point if `checkITFRequest` returns one of the pending status types (`none` or `expired`), in which case, we create a new ITF.